src/shogun/io/SGIO.h

```diff
@@ -328,89 +300,45 @@ class SGIO
 		void done();
 
 		/** print error message 'not implemented' */
-		inline void not_implemented(const char* function, const char* file, int32_t line) const
+		inline void not_implemented(const SourceLocation& loc={}) const
 		{
 			error<ShogunException>(
-			    MSG_ERROR, function, file, line,
-			    "Sorry, not yet implemented .\n");
+			    MSG_ERROR, loc, "Sorry, not yet implemented .\n");
 		}
 
 		/** print error message 'Only available with GPL parts.' */
-		inline void gpl_only(const char* function, const char* file, int32_t line) const
+		inline void gpl_only(const SourceLocation& loc={}) const
 		{
 			error<ShogunException>(
-			    MSG_ERROR, function, file, line, "This feature is only "
-			                                     "available if Shogun is built "
-			                                     "with GPL codes.\n");
+			    MSG_ERROR, loc,
+			    "This feature is only "
+			    "available if Shogun is built "
+			    "with GPL codes.\n");
```

Comment:
the cmake error on this was recently updated by @sbrice .. maybe it should be updated here as well?